storage.go

```diff
@@ -5,1010 +5,1013 @@
-	if err != nil {
-		return err
-	}
-
-	storage := nats.FileStorage
-	if memory {
-		storage = nats.MemoryStorage
-	}
-
-	kv, err := js.KeyValue(ConfigBucketName)
-	if err == nats.ErrBucketNotFound {
-		kv, err = js.CreateKeyValue(&nats.KeyValueConfig{
-			Bucket:      ConfigBucketName,
-			Description: "Choria Async Jobs Configuration",
-			Storage:     storage,
-			Replicas:    replicas,
-		})
-	}
-	if err != nil {
-		return err
-	}
-
-	s.configBucket = kv
-
-	kv, err = js.KeyValue(LeaderElectionBucketName)
-	if err == nats.ErrBucketNotFound {
-		kv, err = js.CreateKeyValue(&nats.KeyValueConfig{
-			Bucket:      LeaderElectionBucketName,
-			Description: "Choria Async Jobs Leader Elections",
-			Storage:     storage,
-			Replicas:    replicas,
-			TTL:         10 * time.Second,
-		})
-	}
-	if err != nil {
-		return err
-	}
-
-	s.leaderElections = kv
-
-	return nil
+  var err error
+
+  if replicas == 0 {
+    replicas = 1
+  }
+
+  js, err := s.nc.JetStream()
+  if err != nil {
+    return err
+  }
+
+  storage := nats.FileStorage
+  if memory {
+    storage = nats.MemoryStorage
+  }
+
+  kv, err := js.KeyValue(ConfigBucketName)
+  if err == nats.ErrBucketNotFound {
+    kv, err = js.CreateKeyValue(&nats.KeyValueConfig{
+      Bucket:      ConfigBucketName,
+      Description: "Choria Async Jobs Configuration",
+      Storage:     storage,
+      Replicas:    replicas,
+      MaxBytes:    DefaultMaxBytes,
+    })
+  }
+  if err != nil {
+    return err
+  }
+
+  s.configBucket = kv
+
+  kv, err = js.KeyValue(LeaderElectionBucketName)
+  if err == nats.ErrBucketNotFound {
+    kv, err = js.CreateKeyValue(&nats.KeyValueConfig{
+      Bucket:      LeaderElectionBucketName,
+      Description: "Choria Async Jobs Leader Elections",
+      Storage:     storage,
+      Replicas:    replicas,
+      TTL:         10 * time.Second,
+      MaxBytes:    DefaultMaxBytes,
+    })
+  }
+  if err != nil {
+    return err
+  }
+
+  s.leaderElections = kv
+
+  return nil
 
 }
 
 func (s *jetStreamStorage) PrepareTasks(memory bool, replicas int, retention time.Duration) error {
-	var err error
+  var err error
 
-	if replicas == 0 {
-		replicas = 1
-	}
+  if replicas == 0 {
+    replicas = 1
+  }
 
-	opts := []jsm.StreamOption{
-		jsm.Subjects(TasksStreamSubjects),
-		jsm.MaxMessagesPerSubject(1),
-		jsm.Replicas(replicas),
-		jsm.StreamDescription("Choria Async Jobs Tasks"),
-	}
+  opts := []jsm.StreamOption{
+    jsm.Subjects(TasksStreamSubjects),
+    jsm.MaxMessagesPerSubject(1),
+    jsm.Replicas(replicas),
+    jsm.StreamDescription("Choria Async Jobs Tasks"),
+  }
 
-	if memory {
-		opts = append(opts, jsm.MemoryStorage())
-	} else {
-		opts = append(opts, jsm.FileStorage())
-	}
+  if memory {
+    opts = append(opts, jsm.MemoryStorage())
+  } else {
+    opts = append(opts, jsm.FileStorage())
+  }
 
-	opts = append(opts, jsm.MaxAge(retention))
+  opts = append(opts, jsm.MaxAge(retention))
+  opts = append(opts, jsm.MaxBytes(DefaultMaxBytes))
```

Comment:
Not sure I am comfortable adding this maximum always, wrap it in a check for either being set specifically or if its required, check for required using this. To detrmine if its set you can use the `IsSetByUser()` support when adding the flag

https://github.com/nats-io/jsm.go/blob/78b71b6a1d19e714732a57a7d130bb697b7ce522/manager.go#L94

Reply:
Example use here:

```
		f.Flag("subscriptions", "Maximum allowed subscriptions").Default("-1").IsSetByUser(&c.maxSubIsSet).Int64Var(&c.maxSubs)
```

this will set the boolean true if the user set it